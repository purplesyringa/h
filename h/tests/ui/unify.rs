fn main() {
    let array = h::map! {
        [1, 2_u32, 3_u64] => ()
    };

    let keys = h::map! {
        1 => (),
        2_u32 => (),
        3_u64 => (),
    };

    let unsizing_cast = h::map! {
        &[1_i32, 2, 3] as &[_] => (),
        &[1_i64, 2, 3, 4] as &[_] => (),
    };

    let what = h::map! {
        1 as &str => (),
    };
}
