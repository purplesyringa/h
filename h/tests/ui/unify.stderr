error: type `u64`, inferred from here, contradicts an earlier type annotation
 --> tests/ui/unify.rs:3:20
  |
3 |         [1, 2_u32, 3_u64] => ()
  |                    ^^^^^

error: the type was inferred as `u32` here
 --> tests/ui/unify.rs:3:13
  |
3 |         [1, 2_u32, 3_u64] => ()
  |             ^^^^^

error: type `u64`, inferred from here, contradicts an earlier type annotation
 --> tests/ui/unify.rs:9:9
  |
9 |         3_u64 => (),
  |         ^^^^^

error: the type was inferred as `u32` here
 --> tests/ui/unify.rs:8:9
  |
8 |         2_u32 => (),
  |         ^^^^^

error: type `i64`, inferred from here, contradicts an earlier type annotation
  --> tests/ui/unify.rs:14:11
   |
14 |         &[1_i64, 2, 3, 4] as &[_] => (),
   |           ^^^^^

error: the type was inferred as `i32` here
  --> tests/ui/unify.rs:13:11
   |
13 |         &[1_i32, 2, 3] as &[_] => (),
   |           ^^^^^

error: cannot cast a value of type `{integer}` to `&_`
       `h` is not a full-blown interpreter; only trivial and unsizing casts (`&[T; N]` to `&[T]`) are supported
  --> tests/ui/unify.rs:18:9
   |
18 |         1 as &str => (),
   |         ^
